transaction/std/src/transaction_builder.rs

```diff
@@ -63,13 +66,21 @@ impl TransactionBuilder {
         recipient: &PublicAddress,
         recipient_fog_ingest_key: Option<&RistrettoPublic>,
         rng: &mut RNG,
-    ) -> Result<TxOut, TxBuilderError> {
+    ) -> Result<(TxOut, TxOutConfirmationNumber), TxBuilderError> {
         let (tx_out, shared_secret) =
             create_output(value, recipient, recipient_fog_ingest_key, rng)?;
 
         self.outputs_and_shared_secrets
             .push((tx_out.clone(), shared_secret));
-        Ok(tx_out)
+
+        let mut hasher = Blake2b256::new();
```

Comment:
It might be worth moving this functionality into a constructor like `TxOutConfirmationNumber::new(shared_secret)`. That would make the intended use of the type clearer, and would avoid duplicate code later when the receiver needs to compute the same thing.

Reply:
`impl From` is probably the right way to do it

Reply:
Done @mfaulk .

Reply:
Weird @eranrund github didn't show me your comment.  I'm happy to do either.